pkg/kubelet/kubelet.go

```diff
@@ -297,6 +297,8 @@ func makePodSourceConfig(kubeCfg *kubeletconfiginternal.KubeletConfiguration, ku
 	if kubeCfg.StaticPodPath != "" {
 		klog.InfoS("Adding static pod path", "path", kubeCfg.StaticPodPath)
 		config.NewSourceFile(kubeCfg.StaticPodPath, nodeName, kubeCfg.FileCheckFrequency.Duration, cfg.Channel(ctx, kubetypes.FileSource))
+	} else {
+		klog.InfoS("There is NO static pod path defined, which may interfere with kubeadm or other K8s bootstrapping functionality - check kubelet configuration file if you want to fix this")
```

Comment:
non blocking comment

kubeadm always creates /etc/kubernetes/manifests (hardcoded) and passes it to the kubelet via config, though.
so the only way this can happen is due to operator error, where they deleted the dir during kubeadm runtime or overrode the config value.

Reply:
I hope that's the case. It appears to be pretty easy to mess this up though.   

...

How can kubeadm force the kubelet to use or not use a specific manifests directory? 
....

Reply:
there is no way to force it per-se, but using a non-standard path will break kubeadm runtime.

- kubeadm has a constant for the /etc/kubernetes/manifests dir
- it passes it to kubeletconfiguration as a field
- it uses it everywhere for manifest management logic
- the user can use the kubelet flag to override this dir on the kubelet side (as kubelet flags override config fields), but then kubeadm will just manage manifests in the known / hardcoded dir and the kubelet will not pick up any manifests.